services/frontend-service/src/ui/components/ReleaseDialog/ReleaseDialog.tsx

```diff
@@ -445,10 +446,82 @@ export const EnvironmentGroupLane: React.FC<{
     // all envs in the same group have the same priority
     const priorityClassName = getPriorityClassName(environmentGroup);
     const [isCollapsed, setIsCollapsed] = React.useState(false);
+    const [allowDeployment, setAllowDeployment] = React.useState(true);
     const collapse = useCallback(() => {
         setIsCollapsed(!isCollapsed);
     }, [isCollapsed]);
 
+    const allReleases = useCurrentlyDeployedAtGroup(app, release.version).filter(
+        (releaseEnvGroup) => releaseEnvGroup.environmentGroupName === environmentGroup.environmentGroupName
+    );
+
+    const createEnvGroupLock = React.useCallback(() => {
+        environmentGroup.environments.forEach((environment) => {
+            addAction({
+                action: {
+                    $case: 'createEnvironmentApplicationLock',
+                    createEnvironmentApplicationLock: {
+                        environment: environment.name,
+                        application: app,
+                        lockId: '',
+                        message: '',
+                        ciLink: '',
+                    },
+                },
+            });
+        });
+    }, [environmentGroup, app]);
+    const deployAndLockClick = React.useCallback(() => {
+        environmentGroup.environments.forEach((environment) => {
+            if (
+                allReleases &&
+                allReleases.length !== 0 &&
+                allReleases[0].environments.find((env) => env === environment)
+            ) {
+                return;
+            }
+            addAction({
+                action: {
+                    $case: 'deploy',
+                    deploy: {
+                        environment: environment.name,
+                        application: app,
+                        version: release.version,
+                        ignoreAllLocks: false,
+                        lockBehavior: LockBehavior.IGNORE,
+                    },
+                },
+            });
+            addAction({
+                action: {
+                    $case: 'createEnvironmentApplicationLock',
+                    createEnvironmentApplicationLock: {
+                        environment: environment.name,
+                        application: app,
+                        lockId: '',
+                        message: '',
+                        ciLink: '',
+                    },
+                },
+            });
+        });
+    }, [release.version, app, environmentGroup, allReleases]);
+
+    React.useEffect(() => {
+        if (allReleases === undefined) {
+            setAllowDeployment(true);
+            return;
+        }
+
+        if (allReleases.length === 0) {
+            setAllowDeployment(true);
+        } else {
+            setAllowDeployment(
+                JSON.stringify(allReleases[0].environments) !== JSON.stringify(environmentGroup.environments)
+            );
+        }
```

Comment:
this seems more complicated than necessary, or maybe I don't understand the goal. Why is it ever necessary to disable the expand button?
But assuming it is necessary:
Please rename "allowDeployment" to something like "enableExpandGroup".

Reply:
Oh, we are not talking about "the expand button" but the deployment button for a group.
In that case maybe "allowGroupDeployment" as a name?